bin/src/hickory-dns.rs

```diff
@@ -186,6 +204,18 @@ fn run() -> Result<(), String> {
 
     let config = Config::read_config(config_path)
         .map_err(|err| format!("failed to read config file from {config_path:?}: {err}"))?;
+
+    #[cfg(feature = "prometheus-metrics")]
+    if !args.disable_prometheus && !config.disable_prometheus() {
+        let socket = args
+            .prometheus_listen_addr
+            .unwrap_or(config.prometheus_listen_addr());
+        // setup tracing/metrics integration and prometheus endpoint
+        config_prometheus(socket)?;
```

Comment:
I noticed in the docs that `PrometheusBuilder::install()` dynamically chooses between spawning its HTTP server task on an existing Tokio runtime, if available, or spawning a new thread, and running a single-threaded Tokio runtime in it. Since we're already setting up a multithreaded runtime, I think it would make sense to use that, both for overhead and latency reasons. We should move this down below the runtime initialization, and then change this line as follows.
```suggestion
        runtime.block_on(async { config_prometheus(socket) })?;
```